packages/db/src/artifacts/json/schema.ts

```diff
@@ -170,8 +170,24 @@ const translations = [
                   }
                 },
 
+                // remove this and just compress into `required`
+                anyOf: undefined,
+
                 // ensure `type` is in array of required property names
-                required: Array.from(new Set([...itemType.required, "type"]))
+                required: Array.from(
+                  new Set([
+                    ...// required may be defined in `"anyOf"` - just take what's
+                    // in common amongst all the anyOf and make sure it has
+                    // `"type"`
+                    (itemType.required ||
+                      itemType.anyOf
+                        .map(option => new Set(option.required))
+                        .reduce((a, b) =>
+                          !a ? b : [...a].filter(x => b.has(x))
+                        )),
+                    "type"
+                  ])
+                )
```

Comment:
For context: the module containing this change is used to convert the JSON-Schema into GraphQL.

We'll soon be removing this (since Truffle DB won't need a GraphQL version of this schema anymore), so the approach I've taken is just to squash all the `anyOf`s via intersection into a single `required` property, calling everything that's not always required as just plain nullable